pkg/storage/store.go

```diff
@@ -200,58 +198,34 @@ func shouldUseBoltDBIndexGatewayClient(cfg Config) bool {
 	return true
 }
 
-func (s *store) storeForPeriod(p config.PeriodConfig, chunkClient client.Client, f *fetcher.Fetcher) (stores.ChunkWriter, stores.Index, func(), error) {
+func (s *store) storeForPeriod(p config.PeriodConfig, chunkClient client.Client, f *fetcher.Fetcher) (stores.ChunkWriter, series.IndexStore, func(), error) {
 	indexClientReg := prometheus.WrapRegistererWith(
 		prometheus.Labels{"component": "index-store-" + p.From.String()}, s.registerer)
 
 	if p.IndexType == config.TSDBType {
-		var (
-			nodeName = s.cfg.TSDBShipperConfig.IngesterName
-			dir      = s.cfg.TSDBShipperConfig.ActiveIndexDirectory
-		)
-		tsdbMetrics := tsdb.NewMetrics(indexClientReg)
 		objectClient, err := NewObjectClient(s.cfg.TSDBShipperConfig.SharedStoreType, s.cfg, s.clientMetrics)
 		if err != nil {
 			return nil, nil, nil, err
 		}
 
-		shpr, err := indexshipper.NewIndexShipper(
-			s.cfg.TSDBShipperConfig,
-			objectClient,
-			s.limits,
-			tsdb.OpenShippableTSDB,
-		)
-		if err != nil {
-			return nil, nil, nil, err
-		}
-		tsdbManager := tsdb.NewTSDBManager(
-			nodeName,
-			dir,
-			shpr,
-			p.IndexTables.Period,
-			util_log.Logger,
-			tsdbMetrics,
-		)
-		// TODO(owen-d): Only need HeadManager
-		// on the ingester. Otherwise, the TSDBManager is sufficient
-		headManager := tsdb.NewHeadManager(
-			util_log.Logger,
-			dir,
-			tsdbMetrics,
-			tsdbManager,
-		)
-		if err := headManager.Start(); err != nil {
-			return nil, nil, nil, err
-		}
-		idx := tsdb.NewIndexClient(headManager, p)
-		writer := tsdb.NewChunkWriter(f, p, headManager)
+		// ToDo(Sandeep): Avoid initializing writer when in read only mode
+		writer, idx, err := tsdb.NewStore(s.cfg.TSDBShipperConfig, p, f, objectClient, s.limits, indexClientReg)
```

Comment:
I don't see `tsdb.NewStore` implemented yet, is that right? I'm guessing we'll just copy this over to the new function (we can also add a check on whether to enable the head manager as well).

Reply:
Sorry, I missed pushing the file. I have pushed it yesterday.